core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1659,9 +1659,14 @@ private[spark] object Utils extends Logging {
   val windowsDrive = "([a-zA-Z])".r
 
   /**
-   * Format a Windows path such that it can be safely passed to a URI.
+   * Format a path such that it can be safely passed to a URI.
    */
-  def formatWindowsPath(path: String): String = path.replace("\\", "/")
+  def formatPath(path: String, windows: Boolean): String = {
+    val formatted = path.replace(" ", "%20")
```

Comment:
Instead of this, how about:

    scala> new URI(null, "/a b c", null)
    res6: java.net.URI = /a%20b%20c